www/js/metrics.js

```diff
@@ -301,69 +420,127 @@ angular.module('emission.main.metrics',['nvd3', 'emission.services', 'ionic-date
           var userCarbonData = getSummaryDataRaw(results[3].user_metrics, 'distance');
-          $scope.carbonData.userCarbon = [];
+          var optimalDistance = getOptimalFootprintDistance(results[3].user_metrics);
+          var worstDistance = getWorstFootprintDistance(results[3].user_metrics);
+          var date1 = $scope.selectCtrl.fromDateTimestamp;
+          var date2 = $scope.selectCtrl.toDateTimestamp;
+          var duration = moment.duration(date2.diff(date1));
+          var days = duration.asDays();
+          //$scope.ca2020 = 43.771628 / 5 * days; // kg/day
+          $scope.carbonData.ca2035 = Math.round(40.142892 / 5 * days) + ' kg CO₂'; // kg/day
+          $scope.carbonData.ca2050 = Math.round(8.28565 / 5 * days) + ' kg CO₂';
+          //$scope.carbonData.userCarbon = [];
           for (var i in userCarbonData) {
-            $scope.carbonData.userCarbon.push({key: userCarbonData[i].key, values: FootprintHelper.getFootprint(userCarbonData[i].values, userCarbonData[i].key)});
+            //$scope.carbonData.userCarbon.push({key: userCarbonData[i].key, values: FootprintHelper.getFootprint(userCarbonData[i].values, userCarbonData[i].key)});
             if (userCarbonData[i].key === "IN_VEHICLE") {
               $scope.carbonData.userVehicleRange = FootprintHelper.getFootprintRaw(userCarbonData[i].values, userCarbonData[i].key);
+              $scope.carbonData.userCarbon = FootprintHelper.getFootprint(userCarbonData[i].values, userCarbonData[i].key);              
+              $scope.carbonData.optimalCarbon = FootprintHelper.getFootprint(optimalDistance, userCarbonData[i].key);
+              $scope.carbonData.worstCarbon = FootprintHelper.getFootprint(worstDistance, userCarbonData[i].key)
+              if(first){
+                lastWeekCarbon = $scope.carbonData.userCarbon;
+                first = false; //If there is data from last week store the data only first time
+              }
+            $scope.carbonData.lastWeekUserCarbon = lastWeekCarbon;
             }
           }
         }
         if (results[3].aggregate_metrics) {
           var aggrCarbonData = getAvgSummaryDataRaw(results[3].aggregate_metrics, 'distance');
-          $scope.carbonData.aggrCarbon = [];
           for (var i in aggrCarbonData) {
-            $scope.carbonData.aggrCarbon.push({key: aggrCarbonData[i].key, values: FootprintHelper.getFootprint(aggrCarbonData[i].values, aggrCarbonData[i].key)});
             if (aggrCarbonData[i].key === "IN_VEHICLE") {
               $scope.carbonData.aggrVehicleRange = FootprintHelper.getFootprintRaw(aggrCarbonData[i].values, aggrCarbonData[i].key);
+              $scope.carbonData.aggrCarbon = FootprintHelper.getFootprint(aggrCarbonData[i].values, aggrCarbonData[i].key);
             }
           }
         }
+        $scope.summaryData.defaultSummary = $scope.summaryData.userSummary;
+        var change = "";
+        var lastWeekCarAndTrain = lastWeekCarbon.replace(/ /g,'').split("~");
```

Comment:
Is this really the best way to do this? string manipulation? since we're calculating the values locally, can't we just keep them as integers? In general, you want to avoid string manipulation (with regular expressions!) if you can because:
a) it is expensive to convert back and forth
b) it introduces unnecessary fragility where it is not necessary